test/util/update-node-workspace-packages-dependencies.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  updateNodeWorkspacePackagesDependencies,
+  PathPackages,
+} from '../../src/util/update-node-workspace-packages-dependencies';
+import {describe, it} from 'mocha';
+import {expect} from 'chai';
+
+describe('updateNodeWorkspacePackagesDependencies', () => {
+  it('updates package dependencies', async () => {
+    function getPackages(updated: boolean): PathPackages {
```

Comment:
I'm still thinking about where/when this code is called. Somehow the aggregate releaser gets all the package.json contents for each package from the individual releaser (which has already bumped the top level version) and passes them through this lerna wrapper before including them in the PR.

> enough information to run the lerna tree resolution 

I think I have a bug in this implementation: I'm ignoring the workspace configuration and defining the entire workspace only from the `PathPackages` input. Instead I need to load up all the package versions (currently being updated or not) so that any package being currently updated can have it's dependencies set to the latest available whether that dependency is being updated in this run or not. I'll add tests for this, e.g:


**release1**: all packages version and dependencies bump
```
pkgA 1.1.1 => 1.1.2
pkgB 1.1.1 => 1.1.2
- pkgA "^1.1.1" => "^1.1.2"
pkgC 1.1.1 => 1.1.2
- pkgB "^1.1.1" => "^1.1.2"
```

**release2**: A and C version bumps
```
pkgA 1.1.2 => 1.1.3
// pkgB unchanged since it isn't part of this release
pkgC 1.1.2 => 1.1.3
- // pkgB dep unchanged
```

**release3**:  B version bump
```
pkgB 1.1.2 => 1.1.3
- pkgA "^1.1.2" => "^1.1.3"  // pkgA dep must be updated to the latest even though A isn't part of this release.
// pkgC unchanged since it isn't part of this release
```

**release4**: A version bump
```
pkgA 1.1.3 => 1.1.4
// pkgB, pkgC unchanged since they aren't part of this release.
```

**release5**:
```
pkgB 1.1.3 => 1.1.4
- pkgA "^1.1.3" => "^1.1.4"  // pkgA dep must be updated to the latest even though A isn't part of this release.
pkgC 1.1.3 => 1.1.4
- pkgB "^1.1.2" => "^1.1.4"
```